### DISCUSSION TITLE
Change rate-per-second based on some external oracle

### DISCUSSION THREAD

**Discussion by razgraf** [sablier-labs/flow#35] [General]
Some users have asked us about adjusting the streaming rate (basically increasing or decreasing the amount streamed) based on some external oracle. This should already be possible manually (e.g. increasing the salary of a user with `_adjustRatePerSecond`) but I was wondering what we could suggest/implement that would bring some sort of automated adjusting into play, including something like an optimistic oracle.

Examples could range from [UMA's](https://docs.uma.xyz/developers/quick-start) KPI options to more customizable logic. Is there an example (even purely from a marketing perspective) we could do to showcase such an integration? Like setting up a grant with a freelancer on [Bountycaster](https://www.bountycaster.xyz/) and increasing or canceling their stream based on delivery/performance?

@sablier-labs/everybody

**Comment by PaulRBerg** [comment]
Possible duplicate of https://github.com/sablier-labs/company-discussions/discussions/46

**Reply by razgraf** [reply]
Yes, you're right, it's a duplicate. Closing and referencing there.
